src/components/buttons/ButtonIcon.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { ComponentPropsWithoutRef, forwardRef, MouseEvent, useMemo } from 'react';
+import { IconProps } from '../icons';
+import { SpinnerOverrideColor } from '../spinner/SpinnerOverrideColor';
+import { Tooltip } from '../tooltip';
+import { useButtonClickHandler } from './Button';
+import { BUTTON_VARIETY_STYLES, BUTTONICON_DIMENSIONS, ButtonIconStyled } from './ButtonStyles';
+import { ButtonSize, ButtonVariety, HTMLButtonProps } from './ButtonTypes';
+
+type TooltipProps = ComponentPropsWithoutRef<typeof Tooltip>;
+type TooltipOptions = Omit<TooltipProps, 'children' | 'content' | 'key'>;
+
+export interface ButtonIconProps extends HTMLButtonProps {
+  Icon: React.ForwardRefExoticComponent<IconProps & React.RefAttributes<HTMLSpanElement>>;
+  'aria-label': string;
```

Comment:
Should be `ariaLabel`, should we also add a `ariaLabelledBy`?

Reply:
that would be confusing, I think: Since we expose `HTMLButtonProps`, aria-label is already available, I'm just making it mandatory here.
Also, ariaLabelledBy would be weird: it means there's a label element related to the button. I think in that case the button would be a standard one, not a label + icon button

Reply:
1. HTMLButtonProps doesn't expose aria-label, it only exposes `'form' | 'id' | 'name' | 'role' | 'style' | 'type'`
2. yeah indeed that makes sense for the ariaLabelledby

Reply:
HTMLButtonProps extend the ARIA props, don't they?

Reply:
We defined it like this: 
```
export type HTMLButtonProps = Pick<
  ButtonHTMLAttributes<HTMLButtonElement>,
  'form' | 'id' | 'name' | 'role' | 'style' | 'type'
>;
```

Reply:
`HTMLButtonProps` is not a native type but a custom one we created. Maybe it's too close to real typing name, feel free to rename it to make it more obvious.

Reply:
interface ButtonHTMLAttributes<T> extends HTMLAttributes<T>
interface HTMLAttributes<T> extends AriaAttributes, DOMAttributes<T>
and AriaAttributes include `"aria-label"?: string | undefined;`

Reply:
OK my bad, I thought it was the native one

Reply:
Same here, totally got confused

Reply:
Ok so please rename it to something less confusing